readme.md

```diff
@@ -660,6 +660,30 @@ As of version `5.0.0` it uses source maps to report coverage for your actual cod
 
 ## FAQ
 
+### How do I test HTTP endpoints?
+
+AVA doesn't have an official assertion library for endpoints, but a great option is [`supertest-as-promised`](https://www.npmjs.com/package/supertest-as-promised).
+Assuming you are using express, you might set up a server for each test:
+```
+let app;
+test.beforeEach(() => {
```

Comment:
Each test file is it's own process (or thread if you want to think of it that way).

`t.context` is shared between your `beforeEach` and the appropriate `test`. So in the above example, even though `beforeEach` is executed 3 times, it gets a unique `t.context` each time that is shared with just a single `test` run. If you are using a closure to pass data between a `beforeEach` and a `test`, that is almost certainly wrong, and should use `t.context` as well.

It's perfectly acceptable to use a closure to pass data created in `t.before` (in fact `t.context` does not work in `t.before`).